src/_shared/services/giphy-api.service.ts

```diff
@@ -32,3 +32,8 @@ export const searchGiphy = async (query: string, limit = 12, offset = 0) => {
   }));
   return items;
 };
+
+export let searchByCategory = (cat: string) => {
+  const items = fetch(SEARCH_URL + '?=' + cat).then((r) => r.json());
```

Comment:
**Issue**: The new function 'searchByCategory' constructs a URL by directly appending the category string to the base URL, which can lead to security vulnerabilities such as URL injection if the category string is not properly sanitized. <br> **Fix**: Ensure that the category string is properly encoded to prevent URL injection attacks. Use 'encodeURIComponent' to encode the category parameter. <br> **Code Suggestion**: 
 ```
 -  const items = fetch(SEARCH_URL + '?=' + cat).then((r) => r.json());
 +  const items = fetch('${SEARCH_URL}?=${encodeURIComponent(cat)}').then((r) => r.json());
 ```